pkg/server/server.go

```diff
@@ -0,0 +1,49 @@
+package server
+
+import (
+	"context"
+	"log/slog"
+	"net/http"
+
+	camelv1alpha "github.com/apache/camel-k/pkg/apis/camel/v1alpha1"
+	"github.com/gin-gonic/gin"
+	"github.com/sco1237896/sco-backend/pkg/client"
+)
+
+var (
+	logger = slog.Default().With(slog.String("component", "server"))
+)
+
+type Options struct {
+	Addr string
+}
+
+func Start(opts Options, cl *client.Client) error {
+	r := setupRouter(cl)
+	logger.Info("starting server")
+	err := r.Run(opts.Addr)
```

Comment:
can we create an issue to enable tls here ? Ideally the sco-operator should leverage cert-manager (on vanilla kube) or service-serving-certificate (on openshift) to encrypt the traffic